zkfacade/src/main/java/com/yahoo/vespa/curator/SingletonManager.java

```diff
@@ -139,6 +139,7 @@ private Task(Type type, SingletonWorker singleton) {
         final Path path;
         final MetricHelper metrics;
         Lock lock = null;
+        Lock toClose = null;
```

Comment:
Would it be clearer to have an addition state on `doom` (say `CLOSING`) to signal this closing state, since that is already used as a state field?

Reply:
(The point is to be able to retry closing the lock, but at the same time not activating, because "we have the lock" (which we don't really have after invalidation).